appserver/tests/payara-samples/test-utils/src/main/java/fish/payara/samples/RemoteDomainRestarter.java

```
@@ -0,0 +1,80 @@
+ * either the CDDL, the GPL Version 2 or to extend the choice of license to
+ * its licensees as provided above.  However, if you add GPL Version 2 code
+ * and therefore, elected the GPL Version 2 license, then the option applies
+ * only if the new code is made subject to such option by the copyright
+ * holder.
+ */
+package fish.payara.samples;
+
+import java.io.IOException;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+
+/**
+ *
+ * @author lprimak
+ */
+public class RemoteDomainRestarter {
+    public static boolean restart() {
+        // first, try remote domain
+        try (CloseableHttpClient httpclient = HttpClients.createDefault()) {
+            String domainURL = System.getProperty("payara.domain.url", "http://localhost:4848");
+            String domainFormat = "%s/__asadmin/%s";
+            HttpGet restartCommand = new HttpGet(String.format(domainFormat, domainURL, "restart-domain"));
+            HttpGet restartRequired = new HttpGet(String.format(domainFormat, domainURL, "_get-restart-required"));
+            try (CloseableHttpResponse httpresponse = httpclient.execute(restartCommand)) {
+                if (httpresponse.getStatusLine().getStatusCode() == HttpStatus.SC_OK) {
+                    Thread.sleep(3 * 1000);
+                    for (int ii = 0; ii < 60 * 100; ++ii) {
+                        Thread.sleep(10);
+                        try (CloseableHttpResponse waitingOnRestart = httpclient.execute(restartRequired)){
+                            if (waitingOnRestart.getStatusLine().getStatusCode() == HttpStatus.SC_OK) {
+                                break;
+                            }
+                        } catch (IOException ex) {
+                        }
```

SonarQube would say: "Undocumented empty block", but I understand. we simply ignore all errors until we receive HTTP 200 ot we are waiting more than 60000 millioseconds + cost of all operations.

---

But when it times out ... I would like to see a stacktrace, at least the last one. Same if the response was other than SC_OK, I want to read it.

---

added javadoc. works as designed